### PR TITLE
feat: column profiling (postgres, redshift)

### LINKED ISSUE #1260: Profiling executor
Soda CL `discover tables` and `profile columns` will both be handled by the profiling executor which, based on a `profiling_cfg` will call upon the correct code-path and metrics.

The executor will also collect the results of either profiling run and take care of uploading them to cloud by leveraging soda-core cloud interaction technology

### PR DESCRIPTION
**Pull request by bastienboutonnet**
Partially addresses: #1258, #1256.
Closes #1260

**NOTE**: we're intentionally merging this in a feature branch rather than main to keep review of the feature incremental. Once all separate PRs are approved and merged into the base branch here that branch itself will be merged to main.

**Big Query and Snowflake** are tackled in #1314 and #1321, respectively

### CODE REVIEW

**Comment by m1n0** [comment]
This looks great! I wont pretend I absorbed the whole profiling logic yet though 😅 

One general remark, but more of an open question - I am a bit concerned that Scan will grow too much and know too much of the domain logic and not be just an orchestrator/runner of everything, I like how the actual run is separated in its own class, but the run methods for profiling and for automated monitoring in scan also contain a bit of logic which IMHO is not ideal. Saying that, I understand that it needs things from scan and data_source_scan so with current state it seems there is no clever way around it.

**Comment by bastienboutonnet** [comment]
Thanks for having a look @m1n0 

Regarding the profiling logic, do you think there is anything I can do to help you understand? Would you prefer someone else to review that part or do we trust it?

For `Scan` becoming too domain specific. I assume you refer to this kind of stuff here: https://github.com/sodadata/soda-core/pull/1307/files#diff-f8f837bc68ee2608ac96d36a4aed01276f8cb946d8cf810ef30247743a5d89c4R404-R436

If so, I also indeed feel like there is a little bit of domain logic in there. I think there should be a way to build a more factory-like thing but I don't really have a perfect and clear idea in mind right now. So I was thinking maybe to get it in as it is now, and make sure it works so we can start allowing some end to end testing and we can take some time to look into whether we can restructure things a bit. 

Do you think involving someone else in the core team would help us in achieving this maybe or should we say let's just try to go back to it a bit later?

**Comment by bastienboutonnet** [comment]
CI should run again once https://github.com/sodadata/soda-core/pull/1316 is merged and this branch is rebased

**Comment by vijaykiran** [comment]
@bastienboutonnet  can you check the tests? The other PR you mentioned is merged.

File: soda/core/tests/helpers/common_test_tables.py [hunk 1]
```diff
@@ -108,3 +108,35 @@
     columns=customers_test_table.columns,
     values=customers_test_table.values,
 )
+
+customers_profiling = TestTable(
```
Comment (tombaeyens) [inline soda/core/tests/helpers/common_test_tables.py:112]: Why this variation?  Why can't the existing test table not be used for this?

**Comment by tombaeyens** [comment]
@vijaykiran Can you review the (dev-)requirements.txt updates?   Are those in line with the package structure we have in place?

File: soda/core/tests/helpers/common_test_tables.py (hunk 1 above)
Comment (bastienboutonnet) [reply soda/core/tests/helpers/common_test_tables.py:112]: it presented some annoying ties between some records and lead to differences in how ordering would be resolved between local and remote dbs although they are the same as well as some painful arbitrary conversions of float into ints. 

So this variation prevent those from happening but is just valid a test as the other ones.

**Pull request merged**